org.osgi.test.assertj.cm/src/main/java/org/osgi/test/assertj/cm/configuration/AbstractConfigurationAssert.java

```diff
@@ -0,0 +1,206 @@
+	 *             equal to the given one.
+	 */
+	public S hasChangeCount(long changeCount) {
+		// check that actual Configuration we want to make assertions on is not
+		// null.
+		isNotNull();
+
+		// overrides the default error message with a more explicit one
+		String assertjErrorMessage = "\nExpecting changeCount of:\n  <%s>\nto be:\n  <%s>\nbut was:\n  <%s>";
+
+		// check
+		long actualChangeCount = actual.getChangeCount();
+		if (actualChangeCount != changeCount) {
+			failWithMessage(assertjErrorMessage, actual, changeCount, actualChangeCount);
+		}
+
+		// return the current assertion for method chaining
+		return myself;
+	}
+
+	/**
+	 * Verifies that the actual Configuration's factoryPid is equal to the given
+	 * one.
+	 *
+	 * @param factoryPid the given factoryPid to compare the actual
+	 *            Configuration's factoryPid to.
+	 * @return this assertion object.
+	 * @throws AssertionError - if the actual Configuration's factoryPid is not
+	 *             equal to the given one.
+	 */
+	public S hasFactoryPid(String factoryPid) {
+		// check that actual Configuration we want to make assertions on is not
+		// null.
+		isNotNull();
+
+		// overrides the default error message with a more explicit one
+		String assertjErrorMessage = "\nExpecting factoryPid of:\n  <%s>\nto be:\n  <%s>\nbut was:\n  <%s>";
+
+		// null safe check
+		String actualFactoryPid = actual.getFactoryPid();
+		if (!Objects.deepEquals(actualFactoryPid, factoryPid)) {
+			failWithMessage(assertjErrorMessage, actual, factoryPid, actualFactoryPid);
+		}
+
+		// return the current assertion for method chaining
+		return myself;
+	}
+
+	/**
+	 * Verifies that the actual Configuration's pid is equal to the given one.
+	 *
+	 * @param pid the given pid to compare the actual Configuration's pid to.
+	 * @return this assertion object.
+	 * @throws AssertionError - if the actual Configuration's pid is not equal
+	 *             to the given one.
+	 */
+	public S hasPid(String pid) {
+		// check that actual Configuration we want to make assertions on is not
+		// null.
+		isNotNull();
+
+		// overrides the default error message with a more explicit one
+		String assertjErrorMessage = "\nExpecting pid of:\n  <%s>\nto be:\n  <%s>\nbut was:\n  <%s>";
+
+		// null safe check
+		String actualPid = actual.getPid();
+		if (!Objects.deepEquals(actualPid, pid)) {
+			failWithMessage(assertjErrorMessage, actual, pid, actualPid);
+		}
+
+		// return the current assertion for method chaining
+		return myself;
+	}
+
+	/**
+	 * Verifies that the actual Configuration's properties is equal to the given
+	 * one.
+	 *
+	 * @param properties the given properties to compare the actual
+	 *            Configuration's properties to.
+	 * @return this assertion object.
+	 * @throws AssertionError - if the actual Configuration's properties is not
+	 *             equal to the given one.
+	 */
+	public S hasProperties(java.util.Dictionary<String, Object> properties) {
```

Comment:
Perhaps this should be `public AbstractDictionaryAssert<?,Map<String,Object>> hasProperiesThat()`, so that you can then use the full power of DictionaryAssert on the properties.

Reply:
we should!